Greet-Dough/src/main/java/utility/Store.java

```diff
@@ -0,0 +1,35 @@
+package utility;
+
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class Store<T> {
```

Comment:
i like this abstract class - i think this encapsulates the basic concept better than the TrackerID